apps/mosaico/src/Main.purs

```diff
@@ -110,31 +120,48 @@ spec ::
                 { response :: ResponseBody
                 , params :: { pageName :: String }
                 }
+          , categoryPage ::
+              GET "/<categoryName>"
+                { response :: ResponseBody
+                , params :: { categoryName :: String }
+                , guards :: Guards ("category" : Nil)
+                }
           , notFound ::
               GET "/<..path>"
                 { response :: ResponseBody
                 , params :: { path :: List String}
                 }
          }
-    , guards :: { credentials :: Maybe UserAuth }
+    , guards ::
+         { credentials :: Maybe UserAuth
+         , category :: Category
+         }
     }
 spec = Spec
 
 main :: Effect Unit
 main = do
-  htmlTemplate <- liftEffect $ FS.readTextFile UTF8 indexHtmlFileLocation
-  let env = { htmlTemplate }
-      handlers =
-        { getDraftArticle: getDraftArticle env
-        , getArticle: getArticle env
-        , assets
-        , tagList: tagList env
-        , frontpage: frontpage env
-        , staticPage: staticPage env
-        , notFound: notFound env Nothing
-        }
-      guards = { credentials: getCredentials }
-  Aff.launchAff_ $ Payload.startGuarded (Payload.defaultOpts { port = serverPort }) spec { handlers, guards }
+  Aff.launchAff_ do
+    htmlTemplate <- liftEffect $ FS.readTextFile UTF8 indexHtmlFileLocation
+    categoryStructure <- Lettera.getCategoryStructure HBL
+    categoryRegex <-
+      -- This is used for matching a category label from a route, such as "/nyheter" or "/norden-och-världen"
+      case Regex.regex "^\\/([\\w|ä|ö|å|-]+)\\b" Regex.ignoreCase of
+        Right r   -> pure r
+        Left _err -> Aff.throwError $ Aff.error "Got weird regex, couldn't parse it. Server is now exploding. Please fix it."
```

Comment:
Or just use some `unsafe` foo to coerce it, having static regex definition string explode is just not likely. Simpler code to not even pattern match.

Reply:
Yes, this is a bit weird. I don't have a clue what kind of input would make this fail. I tried.